sycl/source/detail/queue_impl.hpp

```diff
@@ -406,16 +406,29 @@ class queue_impl {
     handler Handler(Self, MHostQueue);
     Handler.saveCodeLoc(Loc);
     CGF(Handler);
+
     // Scheduler will later omit events, that are not required to execute tasks.
     // Host and interop tasks, however, are not submitted to low-level runtimes
     // and require separate dependency management.
-    if (has_property<property::queue::in_order>() &&
+    bool IsInOrder = has_property<property::queue::in_order>();
```

Comment:
Probably, the property value should be cached to not add much of overhead whilst submitting.